Tests/X509Tests/OCSPPolicyVerifierTests.swift

```diff
@@ -59,41 +59,102 @@ actor TestRequester: OCSPRequester {
+        do {
+            return try await wrapped.query(request: request, uri: uri)
+        } catch {
+            XCTFail("test query closure throw error \(error)", file: file, line: line)
+            throw error
+        }
     }
 }
 
 final class OCSPVerifierPolicyTests: XCTestCase {
-    private static let privateKey = P384.Signing.PrivateKey()
+    private static let ca1Name = try! DistinguishedName {
+        CountryName("US")
+        OrganizationName("Apple")
+        CommonName("Swift Certificate Test CA 1")
+    }
+    private static let ca1PrivateKey = P384.Signing.PrivateKey()
+    
+    private static func ca(ocspServer: String? = nil) -> Certificate {
+        try! Certificate(
+            version: .v3,
+            serialNumber: .init(),
+            publicKey: .init(ca1PrivateKey.publicKey),
+            notValidBefore: Date() - .days(365),
+            notValidAfter: Date() + .days(3650),
+            issuer: ca1Name,
+            subject: ca1Name,
+            signatureAlgorithm: .ecdsaWithSHA384,
+            extensions: Certificate.Extensions {
+                Critical(
+                    BasicConstraints.isCertificateAuthority(maxPathLength: nil)
+                )
+                KeyUsage(keyCertSign: true)
+                SubjectKeyIdentifier(keyIdentifier: ArraySlice(Insecure.SHA1.hash(data: ca1PrivateKey.publicKey.derRepresentation)))
+                if let ocspServer {
+                    AuthorityInformationAccess([
+                        AuthorityInformationAccess.AccessDescription(
+                            method: .ocspServer,
+                            location: GeneralName.uniformResourceIdentifier(ocspServer))
+                    ])
+                }
+            },
+            issuerPrivateKey: .init(ca1PrivateKey)
+        )
+    }
+    private static let ca1: Certificate = ca()
+    
+    fileprivate static let intermediatePrivateKey = P384.Signing.PrivateKey()
+    private static let leafPrivateKey = P384.Signing.PrivateKey()
     private static func certificate(
         subject: DistinguishedName,
+        publicKey: P384.Signing.PublicKey,
         issuer: DistinguishedName,
+        issuerPrivateKey: P384.Signing.PrivateKey,
+        isIntermediate: Bool,
         ocspServer: String? = nil
     ) -> Certificate {
         try! Certificate(
             version: .v3,
             serialNumber: .init(),
-            publicKey: .init(privateKey.publicKey),
+            publicKey: .init(publicKey),
             notValidBefore: Date() - .days(365),
             notValidAfter: Date() + .days(3650),
-            issuer: ca1Name,
-            subject: ca1Name,
+            issuer: issuer,
+            subject: subject,
             signatureAlgorithm: .ecdsaWithSHA384,
             extensions: .init {
+                if isIntermediate {
+                    Critical(
+                        BasicConstraints.isCertificateAuthority(maxPathLength: 1)
```

Comment:
Mild nit: this path length should be 0. The path length reflects the number of _intermediate_ certificates in the path after this.